web/themes/new_weather_theme/assets/js/components/HourlyTable.js

```diff
@@ -42,32 +42,52 @@ class HourlyTable extends HTMLElement {
   }
 
   clickScrollRight(wrapper) {
-    const cellsToRight = Array.from(
-      this.querySelectorAll("table tr:last-child td"),
-    ).filter((cell) => {
-      const diff = cell.offsetLeft - wrapper.scrollLeft;
-      return diff > 16;
-    });
+    const containerWidth = wrapper.getBoundingClientRect().width;
+    const firstColumnWidth = this.querySelector("table tr:last-child th").getBoundingClientRect().width;
+    const visibleWidth = containerWidth;
+    const rightSide = wrapper.scrollLeft + visibleWidth;
 
-    if (cellsToRight.length) {
+    const nextCol = Array.from(this.querySelectorAll("table tr:last-child td"))
+          .find(el => {
+            const left = el.offsetLeft;
+            const right = el.offsetLeft + el.offsetWidth;
+            return left <= rightSide && right >= rightSide;
+          });   
+
+    if (nextCol) {
+      let nextLeftPos = nextCol.offsetLeft - firstColumnWidth - 16;
+      if(Math.floor(nextLeftPos) <= Math.floor(wrapper.scrollLeft)){
+        nextLeftPos = nextCol.nextElementSibling.offsetLeft - firstColumnWidth - 16;
+      }
```

Comment:
And if scrolling to this position would actually cause us to scroll back to the left, instead move to the next element and use that as our scroll basis?